smarts/core/vehicle_index.py

```diff
@@ -17,21 +17,35 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
-import math
 import logging
+import functools
 from copy import copy, deepcopy
-from io import StringIO
 from enum import IntEnum
+from io import StringIO
 from typing import NamedTuple
+import uuid
 
 import numpy as np
 import tableprint as tp
 
 from smarts.core import gen_id
+from smarts.core.utils.string import truncate
+
 from .chassis import AckermannChassis, BoxChassis
-from .vehicle import Vehicle
-from .sensors import SensorState
 from .controllers import ControllerState
+from .sensors import SensorState
+from .vehicle import Vehicle
+
+VEHICLE_INDEX_ID_LENGTH = 64
+
+
+def _2id(id_: str):
+    assert len(id_) <= VEHICLE_INDEX_ID_LENGTH, id_
+
+    id_ = id_.zfill(VEHICLE_INDEX_ID_LENGTH)
```

Comment:
David: could add a sentinel (E.g. $) before the padding to ensure IDs that may come with the padding character already don't mask each other.